test/single_file/user_defined_literals.cc

```diff
@@ -0,0 +1,27 @@
+#include <cstdint>
+
+// Number wraps integer, enforcing explicit casting
+struct Number {
+    
+    uint32_t value = {};
+    
+    // Constructors
+    Number() = default;
+    explicit Number(unsigned long long int v) : value(static_cast<uint32_t>(v)) {}
+
+    operator uint32_t() const { return value; }
```

Comment:
I'd maybe remove this line as it's not used or necessary for reproducing the bug, even though it would normally be a useful addition.